### PR TITLE
Adding my Minivan layout

### PR DESCRIPTION
**Pull request by lexworth**
Added my Minivan layout. 

New to this let me know if I'm doing it totally wrong.

### CODE REVIEW

File: keyboards/thevankeyboards/minivan/keymaps/lexworth/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+NKRO_ENABLE = yes           # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
+ifndef QUANTUM_DIR
```
Comment (noroadsleft) [inline keyboards/thevankeyboards/minivan/keymaps/lexworth/rules.mk]: Lines 19-21 should be deleted.

They are leftover code from an old build system.

File: keyboards/thevankeyboards/minivan/keymaps/lexworth/rules.mk [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in 
+#   the appropriate keymap folder that will get included automatically
+#
+BOOTMAGIC_ENABLE = yes      # Virtual DIP switch configuration(+1000)
```
Comment (noroadsleft) [inline keyboards/thevankeyboards/minivan/keymaps/lexworth/rules.mk]: For future reference, the only rules that are *necessary* to include in a keymap-level `rules.mk` file such as this are rules that differ from those set at the keyboard level (`keyboards/thevankeyboards/minivan/rules.mk`).

For you, that would be:

```
CONSOLE_ENABLE = no         # Console for debug(+400)
BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
```

File: keyboards/thevankeyboards/minivan/keymaps/lexworth/keymap.c [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#include QMK_KEYBOARD_H
+
+// Helpful defines
+#define _______ KC_TRNS
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+	[0] = LAYOUT(KC_GESC, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_BSPC, LT(1,KC_TAB), KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, LT(1,KC_QUOT), KC_LSPO, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSPC, KC_LCTL, KC_LALT, KC_LGUI, KC_ENT, KC_SPC, TG(2), LT(3,KC_APP), KC_DEL),
```
Comment (noroadsleft) [inline keyboards/thevankeyboards/minivan/keymaps/lexworth/keymap.c]: A suggestion, but by no means required:

Future edits to your keymap (moving or changing key functions) will be made easier if your layers are formatted to appear visually similar to how the keyboard is laid out, like this:

https://github.com/qmk/qmk_firmware/blob/4a7baaaf50bc9cce3886ff7e1b6455e028d92ab9/keyboards/thevankeyboards/minivan/keymaps/default/keymap.c#L32-L37

**Comment by drashna** [comment]
Awesome, thanks!

**Comment by noroadsleft** [comment]
@lexworth 

For future reference, QMK recommends against making commits to and pull requests from your master branch, because it can lead to merge conflicts on later PRs (like if you change your keymap and want it updated here). It is highly recommended for QMK development - regardless of what is being done or where - to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

If you need any help with this just ask. There's a number of people around who can walk you through this.

**Pull request merged**